### PR TITLE
Improve  bal dist list command

### LINKED ISSUE #181: Improve bal dist list output
**Description:**
We need to do following improvements to current `bal dist list` command

- Limit number of remote distributions in 1.x channel to show last 10 releases, if all the distributions need to be listed `bal dist list -a` should be used. Here we treat 1.0.x, 1.1.x and 1.2.x similar level
- Local distributions we list the full list as previous even without separating channels
-  Release name has been removed and only version number is kept and mark the -latest release in each channel
- Swan lake Preview, Beta, Alpha need to be removed upon releasing Swan lake GA

Sample proposed output

```
$ bal dist list
‘Distributions available locally: 

  1.1.6
  1.2.4
  2201.1.0
* 2201.2.1

Distributions available remotely:

1.* channel

  ... To list all the previous distributions execute `bal dist list -a`
  1.2.11
  1.2.12
  1.2.13
  1.2.14
  1.2.15
  1.2.16
  1.2.17
  1.2.18
  1.2.19
  1.2.20 - latest

Swan Lake channel

  2201.0.0
  2201.1.0
  2201.1.1
  2201.1.2
  2201.1.3
  2201.2.0
* 2201.2.1 -  latest

```

### PR DESCRIPTION
**Pull request by udda1996**
## Purpose
Improve the bal dist list command to show the distribution version using SemVer. This enables adding extra argument to list down the entire distribution list available in a channel.  
## Goals
fixes #181

### CODE REVIEW

File: src/main/java/org/ballerinalang/command/cmd/ListCommand.java [hunk 1]
```diff
@@ -51,6 +51,9 @@
     @CommandLine.Option(names = {"--help", "-h", "?"}, hidden = true)
     private boolean helpFlag;
 
+    @CommandLine.Option(names = {"--all", "-a"}, hidden = true)
+    private boolean allFlag;
```
Comment (keizer619) [inline src/main/java/org/ballerinalang/command/cmd/ListCommand.java:55]: Lets include this in the help file [like this ](https://github.com/ballerina-platform/ballerina-update-tool/blob/master/src/main/resources/cli-help/ballerina-dist-remove.help#L7)
 https://github.com/ballerina-platform/ballerina-update-tool/blob/master/src/main/resources/cli-help/ballerina-dist-list.help

File: src/main/resources/cli-help/ballerina-dist-list.help [hunk 2]
```diff
@@ -4,6 +4,23 @@ NAME
 SYNOPSIS
        bal dist list
 
+OPTIONS
```
Comment (keizer619) [inline src/main/resources/cli-help/ballerina-dist-list.help:7]: @praneesha Can you review these help content?

File: src/main/resources/cli-help/ballerina-dist-list.help [hunk 3]
```diff
@@ -4,6 +4,23 @@ NAME
...
+       -pr, --pre-releases
+          List the distributions available for you to download under pre-releases channel.
```
Comment (praneesha) [inline src/main/resources/cli-help/ballerina-dist-list.help:9]:
```suggestion
          List the distributions that are available for you to download under the pre-releases channel.
```

File: src/main/resources/cli-help/ballerina-dist-list.help [hunk 4]
```diff
@@ -4,6 +4,23 @@ NAME
...
+
+       -a, --all
+          List all the distributions available for you to download under each channel.
```
Comment (praneesha) [inline src/main/resources/cli-help/ballerina-dist-list.help:12]:
```suggestion
          List all the distributions that are available for you to download under each channel.
```

File: src/main/resources/cli-help/ballerina-dist-list.help [hunk 5]
```diff
@@ -4,6 +4,23 @@ NAME
...
+
 DESCRIPTION
        List all the distributions installed in your local environment
        as well as the distributions that are available for you to download.
+
+EXAMPLES
+       List latest distributions under major channels.
```
Comment (praneesha) [inline src/main/resources/cli-help/ballerina-dist-list.help:19]:
```suggestion
       List the latest distributions under the major channels.
```

File: src/main/resources/cli-help/ballerina-dist-list.help [hunk 6]
```diff
@@ -4,6 +4,23 @@ NAME
...
+         $ bal dist list
+
+       List all the distributions under major channels.
```
Comment (praneesha) [inline src/main/resources/cli-help/ballerina-dist-list.help:22]:
```suggestion
       List all the distributions under the major channels.
```

**Review by praneesha — APPROVED** [review]

**Review by keizer619 — APPROVED** [review]

**Pull request merged**